toolchain/check/convert.cpp

```diff
@@ -499,7 +499,9 @@ static auto ConvertStructToStructOrClass(Context& context,
                             dest_elem_fields.size()});
   for (auto [i, dest_field] : llvm::enumerate(dest_elem_fields)) {
     if (dest_field.name_id == SemIR::NameId::Vptr) {
-      // CARBON_CHECK(ToClass, "Only classes should have vptrs.");
+      bool NonConstexprToClass = ToClass;
+      CARBON_CHECK(NonConstexprToClass, "Only classes should have vptrs.");
```

Comment:
Hm. How would you feel about something like this:
```suggestion
      if constexpr (!ToClass) {
        CARBON_FATAL("Only classes should have vptrs.");
      }
```

Reply:
I'm open to it (applied in 26bd460) - both feel a bit awkward, but such is the way of things.